### PR TITLE
Add a default handshake mode to the dtls connector

### PR DESCRIPTION
**Pull request by boaks**
Add a default handshake mode to the dtls connector.
Add the handshake mode "auto".
Expose destination endpoint context for responses, ack and rst.
Intended to provide a specific handshake mode different from the dtls
connector's default handshake mode.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 1]
```diff
@@ -1081,6 +1099,30 @@ public Builder setServerOnly(boolean enable) {
...
+		/**
+		 * Set the <em>DTLSConnector</em> default handshake mode.
+		 * 
+		 * @param defaultHandshakeMode
+		 *            {@link DtlsEndpointContext#HANDSHAKE_MODE_AUTO} or
+		 *            {@link DtlsEndpointContext#HANDSHAKE_MODE_NONE}
+		 * @return this builder for command chaining
+		 */
+		public Builder setDefaultHandshakeMode(String defaultHandshakeMode) {
+			if (config.serverOnly) {
```
[the remark this answers is not in the document]
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: OK, check added.

**Comment by boaks** [comment]
Though unit test are currently anyway missing, would it be possible, that you check, if this works for leshan?

**Comment by sbernard31** [comment]
I have no easy way to test this with Leshan. 
Do you have something specific in mind ? 
I could try to integrate this PR and do so manual tests by tweaking Leshan a bit to try to reproduce the use case.

About the API usage, my plan is to set default mode to NONE either on DTLSConnector or Exchange.
and set the mode for request depending on device binding mode.

**Comment by boaks** [comment]
My intention was mainly to check, if the API fits.
The function itself will be tested by unit test, which will be added a little later.

**Comment by boaks** [comment]
unit tests added.

**Pull request merged**